pkg/generate/config/config.go

```diff
@@ -86,6 +96,14 @@ type ResourceConfig struct {
 	ListOperation *ListOperationConfig `json:"list_operation,omitempty"`
 }
 
+// Custom resource fields paths list that map to custom operation
+type CustomUpdateOperationConfig struct {
+	// string paths to fields on the custom resource
+	// Example:
+	//  For field ReplicationGroup.Spec.ReplicasPerNodeGroup, the string value would be: Spec.ReplicasPerNodeGroup
+	DiffPaths []string `json:"diff_paths"`
```

Comment:
Is there a reason we want DiffPaths to be a slice of string instead of just a string? Are there situations where a single update custom update method would be fired for **multiple** DiffPaths?

If there are no situations like that, probably would be simpler to get rid of `CustomUpdateOperationConfig` entirely and just have `ResourceConfig.CustomUpdateOperations` Go type be `map[string]string` where the map keys are the names of the custom methods on the `resourceManager` and the map values are the DiffPath that triggers that call to a custom method.

Thoughts?

Reply:
There are multiple DiffPaths that can lead to a custom method. For example, replication group create API supports:
```
[--num-node-groups   <value>]
[--replicas-per-node-group   <value>]
```
values for these are numbers and service handles creation of node-groups (shards), replicas in each shard per the provided number. Service handles determining the fine-grained details for shards, replicas itself.

However, replication group create API also supports fine grained option to configure each shard during create:
```
[--node-group-configuration   <value>]
```
here the value is an array where each entry in array provide fine-grained values for the node-groups (shards) and replicas:
```
[
{
    "NodeGroupId":   "string",
    "Slots": "string",
    "ReplicaCount": integer,
    "PrimaryAvailabilityZone":   "string",
    "ReplicaAvailabilityZones":   ["string", ...]
},
...
]
```

Thus, there are multiple fields that can lead to custom update methods.